### PR TITLE
Switch to bundler 2.5.9 and commit top gemfile.lock

### PR DESCRIPTION
**Pull request by justin808**
### Summary
2.5.3 lock of bundler causes these errors for me:

Gem::Ext::BuildError: ERROR: Failed to build gem native extension.

    current directory: /Users/justin/.asdf/installs/ruby/3.2.2/lib/ruby/gems/3.2.0/gems/nio4r-2.5.8/ext/nio4r
/Users/justin/.asdf/installs/ruby/3.2.2/bin/ruby extconf.rb
checking for unistd.h... yes
checking for linux/aio_abi.h... no
checking for linux/io_uring.h... no
checking for sys/select.h... yes
checking for port_event_t in poll.h... no
checking for sys/epoll.h... no
checking for sys/event.h... yes
checking for sys/queue.h... yes
checking for port_event_t in port.h... no
checking for sys/resource.h... yes
creating Makefile

current directory: /Users/justin/.asdf/installs/ruby/3.2.2/lib/ruby/gems/3.2.0/gems/nio4r-2.5.8/ext/nio4r
make DESTDIR\= sitearchdir\=./.gem.20240515-35487-luo5ud sitelibdir\=./.gem.20240515-35487-luo5ud clean

- - -
This change is [Reviewable](https://reviewable.io/reviews/shakacode/react_on_rails/1619)

## Summary by CodeRabbit

- **Chores**
  - Updated Bundler version from 2.5.3 to 2.5.9 across all CI/CD configuration files for improved compatibility and potential bug fixes.
  - Removed specific Bundler version constraint in development dependencies for more flexible gem management.

### CODE REVIEW

**Review by Judahmeek — APPROVED** [review]
LGTM

File: Gemfile.development_dependencies [hunk 1]
```diff
@@ -9,7 +9,6 @@ gem "sass-rails", "~> 6.0"
 gem "uglifier"
 gem "jquery-rails"
 gem "puma", "~> 6.0"
-gem "bundler", "2.5.3"
```
Comment (justin808) [inline Gemfile.development_dependencies:12]: @Judahmeek any need for this?

File: .travis.yml [hunk 2]
```diff
@@ -3,7 +3,7 @@ sudo: required
 language: ruby
 
 rvm:
-  - 2.5.3
```
Comment (justin808) [inline .travis.yml:6]: @Judahmeek I don't think the .travis file is used any more.

**Comment by justin808** [comment]
@Judahmeek if this works, I'll merge.

File: .travis.yml (hunk 2 above)
Comment (Judahmeek) [reply .travis.yml:6]: Neither Travis or CircleCI are used anymore. Would you like us to remove those files?

Comment (justin808) [reply .travis.yml:6]: Circle CI might be a good example to keep. I'd say that we could move the files to an /archive directory with a README? Or maybe one can find in the git history if we document that this in a section on CI for ROR?

**Pull request merged**